### ISSUE TITLE
Clarify the difference between getSeries and getSeriesName

### ISSUE THREAD

**Issue by dandv** [node-influx/node-influx#197]
https://github.com/node-influx/node-influx#getseries

**Comment by connor4312** [comment]
Yep, `getSeries` should be `getMeasurements`. This came up tangentially in another Question issue a few months ago, thanks for sticking it here!

**Comment by dandv** [comment]
There's already [getMeasurements](https://github.com/node-influx/node-influx#getmeasurements), but the data it returns is pretty raw - res[0].series[0].values.

Also the `seriesId` in [dropSeries](https://github.com/node-influx/node-influx#dropseries) should be explained.

**Comment by connor4312** [comment]
Resolved in https://github.com/node-influx/node-influx/commit/4988e5212f5a820a0fa1d780a3e5e90ad19f0531

 **1.** There's no reason to have both getSeries and getSeriesNames. In the code it looks like the intention was to allow getSeries to return the full, original result from Influx. But this result only contains the series names, there's not a reason (right now) to have it as a separate method. In the linked commit, `getSeries` exhibits the behaviour of getSeriesName currently

**2.** `DROP SERIES` is actually a slightly more complicated method that what the 4.x code accounts for with `seriesId`. InfluxQL defines it as 
```
drop_series_stmt = "DROP SERIES" ( from_clause | where_clause | from_clause where_clause ) .
```

In 5.x I have implemented it as follows, taking directly from the example in the docstring. Right now it allows the user to either specify a string or use a light query builder which handles escaping and syntax details for them. Using the query builder is a bit more typing, but using it provides the developer with nice autocompletions (yay TypeScript!) and reduces the chance of error with Influx syntax.

```js
// The following pairs of queries are equivalent: you can chose either to
// use our builder or pass in string directly. The builder takes care
// of escaping and most syntax handling for you.

influx.dropSeries({ where: e => e.tag('cpu').equals.value('cpu8') })
influx.dropSeries({ where: '"cpu" = \'cpu8\'' })
// DROP SERIES WHERE "cpu" = 'cpu8'

influx.dropSeries({ measurement: m => m.name('cpu').policy('autogen') })
influx.dropSeries({ measurement: '"cpu"."autogen"' })
// DROP SERIES FROM "autogen"."cpu"

influx.dropSeries({
  measurement: m => m.name('cpu').policy('autogen')
  where: e => e.tag('cpu').equals.value('cpu8')
})
// DROP SERIES FROM "autogen"."cpu" WHERE "cpu" = 'cpu8'
```

I think this resolves the issue, let me know if you've any other thoughts ;)

**Comment by dandv** [comment]
Where is the query builder documented? I saw a proposal in #16.

**Comment by connor4312** [comment]
It's not documented yet. We should be able to get mostly automatic documentation going via tsdoc; for the 5.0.0 I'm planning to whip up a nice github pages site with rich documentation. For now you can check out the class [here](https://github.com/node-influx/node-influx/blob/5.0.0/src/builder.ts#L7), it's quite simple.

I'm somewhat hesitant to make or classify it as a full-on query builder... more often than not I've found it's easier to write SQL directly (when using sql builders) than try to translate into a query builder, I'm not totally sure that's a direction I want to head. The goal of this is to provide a Mongo-esque way to build queries, which is more natural to most JavaScripters, while avoiding the somewhat messy structures you can sometimes get when trying to create complex expressions. E.g. from Sequelize's docs:

```js
{
  $or: [
    {
      title: {
        $like: 'Boat%'
      }
    },
    {
      description: {
        $like: '%boat%'
      }
    }
  ]
}
// title LIKE 'Boat%' OR description LIKE '%boat%'
```

In the Influx builder this would be expressed more imperatively as:

```js
where: e => e.field('title').matches.value(/Boat/).or.field('description').matches.value(/boat/)
// I should change is so that consumers can do this instead:
where: e => e.field('title').matches(/Boat/).or.field('description').matches(/boat/)
```

Right now it's just an experiment. I might remove it before release if people don't like it or don't feel that it's valuable. Totally open to feedback here :)

**Comment by connor4312** [comment]
Decided to drop the builder in favor of Mongo-esque queries. The builder is neat, but it's better to be unsurprising than neat.

**Comment by dandv** [comment]
I'd also favor the Chai-style assertions. Also something like,

```js
client.query(measurement).where(tag, '==', value).and(time, '>', 'now() - 1h');
```

**Comment by connor4312** [comment]
Hm, other ORMs like Eloquent and Bookshelf do something similar. SQLAlchemy leverages operator overloading to do that, for great fun and magic. That sounds like a good idea to me 😄  I'll make the current miniature query builder part of a full builder, maintained as a separate repo/project.

Existing builders like Knex and Squel exist, I'll see  if leveraging one of those makes sense. InfluxQL may be different enough from SQL that rolling our own would lead to less complexity and greater fluidity.

**Comment by connor4312** [comment]
These changes have been published in `5.0.0-alpha.1`. You can install this via:

```
npm install --save influx@next
```
